### PR TITLE
CIRCLE-16271 apply patch for CVE 2019-5736

### PR DESCRIPTION
**Pull request by bear**
Adjust the cloud init script that installs Docker for CVE 2019-5736

This change uses jq to add a single entry to the installed docker config file

Doesn't assume the presence of jq
Saves the original file to /etc/docker/daemon.json.orig
Assumes:
- the path for the docker config is /etc/docker/daemon.json
- we are able to write to the above file

### CODE REVIEW

File: modules/nomad-cloudinit-ubuntu-v1/templates/nomad_user_data.tpl [hunk 1]
```diff
@@ -24,6 +24,12 @@ apt-get install -y "linux-image-$UNAME"
 apt-get update
 apt-get -y install docker-ce=17.03.2~ce-0~ubuntu-xenial
 
+# force docker to use userns-remap to mitigate CVE 2019-5736
+apt-get -y install jq
+tmp=$(mktemp)
+cp /etc/docker/daemon.json /etc/docker/daemon.json.orig
+jq '."userns-remap": "default"' /etc/docker/daemon.json > "$tmp" && mv "$tmp" /etc/docker/daemon.json
```
Comment (mharriscircleci) [inline modules/nomad-cloudinit-ubuntu-v1/templates/nomad_user_data.tpl]: i'm getting a syntax error testing this jq statement on my machine. The assignment operator in jq is `=`

Comment (bear) [reply modules/nomad-cloudinit-ubuntu-v1/templates/nomad_user_data.tpl]: the syntax error has been fixed

**Comment by bear** [comment]
@dcarley can you give this a final review?

File: modules/nomad-cloudinit-ubuntu-v1/templates/nomad_user_data.tpl [hunk 2]
```diff
@@ -24,6 +24,14 @@ apt-get install -y "linux-image-$UNAME"
 apt-get update
 apt-get -y install docker-ce=17.03.2~ce-0~ubuntu-xenial
 
+# force docker to use userns-remap to mitigate CVE 2019-5736
+apt-get -y install jq
+mkdir -p /etc/docker
+[ -f /etc/docker/daemon.json ] || echo '{}' > /etc/docker/daemon.json
+tmp=$(mktemp)
+cp /etc/docker/daemon.json /etc/docker/daemon.json.orig
+jq '."userns-remap"= "default"' /etc/docker/daemon.json > "$tmp" && mv "$tmp" /etc/docker/daemon.json
```
Comment (ryanwohara) [inline modules/nomad-cloudinit-ubuntu-v1/templates/nomad_user_data.tpl]: This actually throws an error on Trusty using jq 1.3. It does work in 1.5.1 but not 1.3, the latest available to Trusty. Adding some brackets solves it:

```suggestion
jq '.["userns-remap"]="default"' /etc/docker/daemon.json > "$tmp" && mv "$tmp" /etc/docker/daemon.json
```

**Pull request merged**